css-typed-om/Overview.bs

```diff
@@ -3509,58 +3509,140 @@ regardless of what property it is for.
 ::
 
 : <a>font</a>
-::
+:: For both specified and computed values,
+    reify as a {{CSSStyleValue}}
+    and return the result.
 
 : <a>font-family</a>
-::
+:: For both specified and computed values,
+    reify as a {{CSSStyleValue}}
+    and return the result.
 
 : <a>font-language-override</a>
-::
+:: For both specified and computed values:
+
+    1. If the value is <css>normal</css>,
+        [=reify an identifier=] from the value
+        and return the result.
+
+    2. Otherwise, reify as a {{CSSStyleValue}}
+        and return the result.
 
 : <a>font-max-size</a>
-::
+:: For both specified and computed values:
+
+    1. If the value is an <<absolute-size>>, <<relative-size>> or <css>infinity</css>, 
+        [=reify an identifier=] from the value
+        and return the result.
+
+    2. Otherwise, [=reify a numeric value=] from the value
+        and return the result.
 
 : <a>font-min-size</a>
-::
+:: Same as 'font-size'
 
 : <a>font-optical-sizing</a>
-::
+:: For both specified and computed values,
+    [=reify an identifier=] from the value
+    and return the result.
 
 : <a>font-palette</a>
-::
+:: For both specified and computed values:
+
+    1. If the value is <css>normal</css>, <css>light</css> or <css>dark</css>,
+        [=reify an identifier=] from the value
+        and return the result.
+
+    2. Otherwise, reify as a {{CSSStyleValue}}
+        and return the result.
 
 : <a>font-presentation</a>
-::
+:: For both specified and computed values,
+    [=reify an identifier=] from the value
+    and return the result.
 
 : <a>font-size</a>
-::
+:: For both specified and computed values:
+
+    1. If the value is an <<absolute-size>> or <<relative-size>>, 
+        [=reify an identifier=] from the value
+        and return the result.
+
+    2. Otherwise, [=reify a numeric value=] from the value
+        and return the result.
 
 : <a>font-size-adjust</a>
-::
+:: For both specified and computed values:
+
+    1. If the value is <css>none</css>, 
+        [=reify an identifier=] from the value
+        and return the result.
+
+    2. Otherwise, [=reify a numeric value=] from the value
+        and return the result.
 
 : <a>font-stretch</a>
-::
+:: For both specified and computed values:
+
+    1. If the value is a <<percentage>>, 
+        [=reify a numeric value=] from the value
+        and return the result.
+
+    2. Otherwise, [=reify an identifier=] from the value
+        and return the result.
 
 : <a>font-style</a>
-::
+:: For both specified and computed values:
+
+    1. If the value is <css>normal</css> or <css>italic</css>, 
+        [=reify an identifier=] from the value
+        and return the result.
+
+    2. Otherwise, reify as a {{CSSStyleValue}}
+        and return the result.
 
 : <a>font-synthesis</a>
-::
+:: For both specified and computed values:
+
+    1. If the value is <css>none</css>, <css>weight</css>, <css>style</css> or <css>small-caps</css>,
```

Comment:
The syntax is `none | [ weight || style || small-caps ]`. Is this correct? I wonder what will happen when we support values like `weight style`. Might be some backwards compat issues there?

Reply:
Specifically, I mean if we reify `weight style` as a list, then I have to write code like:

```
// Want to know if 'small-caps' is synthesized.
const result = styleMap.get('font-synthesis');
if (result is keyword with value 'small-caps' or result is a list containing value 'small-caps')
```

instead of just
```
if (result is a list containing value 'small-caps')
```

Reply:
No, "weight style" should definitely be a CSSStyleValue; it's a single value with two keywords in it, not a list-valued property.

Reply:
Ah yes that makes sense. We can just rely on the upgrade mechanism here.